src/Symfony/Component/Security/Http/EventListener/LoginThrottlingListener.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\RateLimiter\Limiter;
+use Symfony\Component\Security\Core\Exception\TooManyLoginAttemptsAuthenticationException;
+use Symfony\Component\Security\Http\Authenticator\Passport\Badge\UserBadge;
+use Symfony\Component\Security\Http\Event\CheckPassportEvent;
+use Symfony\Component\Security\Http\Event\LoginSuccessEvent;
+
+/**
+ * @author Wouter de Jong <wouter@wouterj.nl>
+ *
+ * @final
+ * @experimental in 5.2
+ */
+class LoginThrottlingListener implements EventSubscriberInterface
```

Comment:
final?

Reply:
When should we use `@final` and when `final`? IIRC, I've used `@final` for all experimental Security classes and not the PHP `final` keyword. And I remember mostly seeing `@final` in the Symfony code.

(I'm fine with both, but I want to prevent wasting your time with such comments in future PRs of mine)

Reply:
We should only use `@final` for existing classes for which adding `final` would be a BC break. Then, in the next major version, we can convert the annotation to a real final class.